nest-forms-backend/src/ginis/ginis.service.ts

```diff
@@ -265,30 +200,65 @@ export default class GinisService {
     }
   }
 
+  private async updateFailedAssignment(ginisDocumentId: string): Promise<void> {
+    await this.prismaService.forms.update({
+      where: {
+        ginisDocumentId,
+      },
+      data: {
+        state: FormState.ERROR,
+        error: FormError.GINIS_SEND_ERROR,
+        ginisState: GinisState.ERROR_ASSIGN_SUBMISSION,
+      },
+    })
+  }
+
+  private async updateSuccessfulAssignment(
+    ginisDocumentId: string,
+  ): Promise<void> {
+    await this.prismaService.forms.update({
+      where: { ginisDocumentId },
+      data: {
+        ginisState: GinisState.SUBMISSION_ASSIGNED,
+        error: FormError.NONE,
+        state: FormState.PROCESSING,
+      },
+    })
+  }
+
   async assignSubmission(
-    documentId: string,
-    organization: string,
-    person?: string,
+    ginisDocumentId: string,
+    ginisNodeId: string,
+    ginisFunctionId?: string,
   ): Promise<void> {
     this.logger.debug('---- start to assign submission ----')
     await this.prismaService.forms.update({
       where: {
-        ginisDocumentId: documentId,
+        ginisDocumentId,
       },
       data: {
         ginisState: GinisState.RUNNING_ASSIGN_SUBMISSION,
       },
     })
 
-    await this.rabbitMqClientService.publishMessageToGinisAutomation(
-      GINIS_AUTOMATION_ASSIGN_QUEUE,
-      {
-        doc_id: documentId,
-        organization,
-        ...(person ? { person } : {}),
-      },
-      ASSIGN_QUEUE,
-    )
+    try {
+      await this.ginisHelper.retryWithDelay(async () =>
+        this.ginisApiService.assignDocument(
+          ginisDocumentId,
+          ginisNodeId,
+          ginisFunctionId,
+        ),
+      )
+      await this.updateSuccessfulAssignment(ginisDocumentId)
+      this.logger.debug('---- assigned in ginis ----')
```

Comment:
I suppose these logs are taken from the old code? They are kinda useless and weird format, but let's keep it and revamp them during the rewrite.